notifications/bridge.go

```diff
@@ -172,6 +192,36 @@ func (b *bridge) createBlobEvent(action string, repo reference.Named, desc distr
 	return event, nil
 }
 
+func (b *bridge) createTagDeleteEventAndWrite(action string, repo reference.Named, tag string) error {
+	return b.sink.Write(*b.createTagBaseEvent(action, repo, tag, distribution.Descriptor{}))
+}
+
+func (b *bridge) createTagEvent(action string, repo reference.Named, tag string, desc distribution.Descriptor) (*Event, error) {
+	event := b.createTagBaseEvent(action, repo, tag, desc)
+
+	ref, err := reference.WithTag(repo, event.Target.Tag)
+	if err != nil {
+		return nil, err
+	}
+
+	event.Target.URL, err = b.ub.BuildManifestURL(ref)
+	if err != nil {
+		return nil, err
+	}
+	return event, nil
+}
+
+func (b *bridge) createTagBaseEvent(action string, repo reference.Named, tag string, desc distribution.Descriptor) *Event {
+	event := b.createEvent(action)
+	event.Target.Digest = desc.Digest
+	event.Target.Size = desc.Size
+	event.Target.MediaType = desc.MediaType
+	event.Target.Repository = repo.Name()
+	event.Target.Tag = tag
```

Comment:
`Target` is not a type, but an anonymous struct.  `Descriptor` can be assigned directly though.